LearnositySDK/Request/DataApi.cs

```diff
@@ -160,7 +160,7 @@ private JsonObject handleRequestRecursive(string url, JsonObject securityPacket,
                     }
                 }
             }
-            while(Tools.array_key_exists("next", meta));
+            while(Tools.array_key_exists("next", meta) && !Tools.empty(data.getJsonObject("data")));
```

Comment:
@gonzalozawa we've fixed infinite loop in PHP SDK here: https://github.com/Learnosity/learnosity-sdk-php/commit/7c4092eecb616eb658907e4e96561721abf0073e

It's similar to your fix but uses `$requestPacket` as a condition for `while` loop. I suggested to have the same solution in .NET SDK, as it's easier then to compare both SDKs. .NET was written based on PHP one and let's keep it this way.